lm_eval/tasks/math.py

```diff
@@ -0,0 +1,343 @@
+    def download(self):
+        if not os.path.exists('data/'):
+            sh("mkdir data")
+        if not os.path.exists('data/MATH/'):
+            sh("wget https://people.eecs.berkeley.edu/~hendrycks/MATH.tar.gz -P data/")
+            sh("tar -xvf data/MATH.tar.gz -C data/")
+            sh("rm data/MATH.tar.gz")
+        self.set_docs()
+
+    @abc.abstractmethod
+    def get_file_info(self):
+        """returns directory name"""
+        pass
+
+    def set_docs(self):
+        self._training_docs = []
+        self._testing_docs = []
+        dir_name = self.get_file_info()
+        train_path = os.path.join('data/MATH/train',dir_name).replace("\\", "/") 
+        test_path = os.path.join('data/MATH/test',dir_name).replace("\\", "/")
+
+        for filename in os.listdir(train_path):
+            with open(os.path.join(train_path, filename).replace("\\", "/")) as f: 
+                self._training_docs.append(json.load(f))
+        for filename in os.listdir(test_path):
+            with open(os.path.join(test_path, filename).replace("\\", "/")) as f: 
+                self._testing_docs.append(json.load(f))
+
+        for doc in self._testing_docs:
+            doc["answer"] = self.remove_boxed(self.last_boxed_only_string(doc["solution"]))
+
+
+    def has_training_docs(self):
+        return True
+
+    def has_validation_docs(self):
+        return False
+
+    def has_test_docs(self):
+        return True
+
+    def training_docs(self):
+        return self._training_docs
+
+    def validation_docs(self):
+        return NotImplemented
+
+    def test_docs(self):
+        return self._testing_docs
+
+    def fewshot_examples(self, k):
+        assert k <= 8, "There are only 8 possible shots for this task."
+        prompts = [
+            {"problem": "What is $\left(\\frac{7}{8}\\right)^3 \cdot \left(\\frac{7}{8}\\right)^{-3}$?", "answer": "$1$"},
+            {"problem": "In how many ways can 4 books be selected from a shelf of 6 books if the order in which the books are selected does not matter?", "answer": "$15$"},
+            {"problem": "Find the distance between the points $(2,1,-4)$ and $(5,8,-3).$", "answer": "$\sqrt{59}$"},
+            {"problem": "The faces of an octahedral die are labeled with digits $1$ through $8$. What is the probability, expressed as a common fraction, of rolling a sum of $15$ with a pair of such octahedral dice?", "answer": "$\\frac{1}{32}$"},
+            {"problem": "The first three terms of an arithmetic sequence are 1, 10 and 19, respectively. What is the value of the 21st term?" , "answer": "$181$"},
+            {"problem": "Calculate $6 \\cdot 8\\frac{1}{3}", "answer": "$50$"},
+            {"problem": "When the binary number $100101110010_2$ is divided by 4, what is the remainder (give your answer in base 10)?", "answer": "$2$"},
+            {"problem": "How many zeros are at the end of the product 25 $\\times$ 240?", "answer": "$3$"}
+        ]
+        return prompts[:k]
```

Comment:
We should probably do the thing we did for https://github.com/EleutherAI/lm-evaluation-harness/pull/154 where we have separate tasks for the "original" version and the modified few-shot version